lib/keyfile/keyfile.c

```diff
@@ -122,6 +123,9 @@ read_encrypted(const uint8_t * keybuf, size_t keylen, uint64_t * machinenum,
 	size_t deckeylen;
 	int rc;
 
+	/* It doesn't make sense to read an empty filename. */
+	assert(filename != NULL);
```

Comment:
NULL != "empty".  I'd go with something along the lines of "The caller must pass a file name to be read."